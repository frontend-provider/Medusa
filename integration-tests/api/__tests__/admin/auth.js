const path = require("path")
const { Region, DiscountRule, Discount } = require("@medusajs/medusa")

const setupServer = require("../../../helpers/setup-server")
const { useApi } = require("../../../helpers/use-api")
const { initDb, useDb } = require("../../../helpers/use-db")
const adminSeeder = require("../../helpers/admin-seeder")
const { exportAllDeclaration } = require("@babel/types")

jest.setTimeout(30000)

describe("/admin/auth", () => {
  let medusaProcess
  let dbConnection

  beforeAll(async () => {
    const cwd = path.resolve(path.join(__dirname, "..", ".."))
    dbConnection = await initDb({ cwd })
    medusaProcess = await setupServer({ cwd })

    await adminSeeder(dbConnection)
  })

  afterAll(async () => {
    const db = useDb()
    await db.shutdown()
    medusaProcess.kill()
  })

  it("creates admin session correctly", async () => {
    const api = useApi()

    const response = await api
      .post("/admin/auth", {
        email: "admin@medusa.js",
        password: "secret_password",
      })
      .catch((err) => {
        console.log(err)
      })

    expect(response.status).toEqual(200)
    expect(response.data.user.password_hash).toEqual(undefined)
    expect(response.data.user).toMatchSnapshot({
      email: "admin@medusa.js",
      created_at: expect.any(String),
      updated_at: expect.any(String),
    })
  })
})
