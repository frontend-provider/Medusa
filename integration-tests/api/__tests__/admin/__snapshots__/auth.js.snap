// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/admin/auth creates admin session correctly 1`] = `
Object {
  "api_token": "test_token",
  "created_at": Any<String>,
  "deleted_at": null,
  "email": "admin@medusa.js",
  "first_name": null,
  "id": "admin_user",
  "last_name": null,
  "metadata": null,
  "role": "admin",
  "updated_at": Any<String>,
}
`;
